lib/bolt/shell/bash.rb

```diff
@@ -390,14 +389,25 @@ def execute(command, sudoable: false, **options)
           # See if we can read from out or err, or write to in
           ready_read, ready_write, = select(read_streams.keys, write_stream, nil, timeout)
 
-          # Read from out and err
           ready_read&.each do |stream|
+            stream_name = stream == out ? 'out' : 'err'
             # Check for sudo prompt
-            read_streams[stream] << if use_sudo
-                                      check_sudo(stream, inp, options[:stdin])
-                                    else
-                                      stream.readpartial(CHUNK_SIZE)
-                                    end
+            to_print = if use_sudo
+                         check_sudo(stream, inp, options[:stdin])
+                       else
+                         stream.readpartial(CHUNK_SIZE)
+                       end
+
+            if !to_print.chomp.empty? && @stream_logger
+              formatted = ""
+              to_print.split("\\n").each_with_index do |msg, i|
+                indent = i == 0 ? '' : '  '
+                formatted += "[#{@target.safe_name}] #{stream_name}: #{indent}#{msg.chomp}\n"
```

Comment:
Not super sure about the indent here, mostly because it's not really noticeable with just two spaces. Maybe just the source being added to each line is enough?

Reply:
Yeah - the spec example is especially weird since it's JSON. I honestly don't mind it but I can definitely see how it's confusing rather than clarifying. We can always reformat if The People ask for it.